src/net.h

```diff
@@ -729,7 +729,7 @@ class CNode
 
     // flood relay
     std::vector<CAddress> vAddrToSend;
-    CRollingBloomFilter addrKnown;
+    std::unique_ptr<CRollingBloomFilter> m_addr_known;
```

Comment:
```suggestion
    const std::unique_ptr<CRollingBloomFilter> m_addr_known;
```

in the first commit:

On a second thought, this should be made `const` and `m_addr_relay_peer` should be removed:

* Slightly less code
* Easier to review, because equivalence of `m_addr_relay_peer == bool{m_addr_known}` is obviously true
* 4 bytes less memory per peer

Reply:
For the full proposed diff:

```diff
diff --git a/src/net.cpp b/src/net.cpp
index 3cea184ac3..84692d2a79 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -2669,7 +2669,7 @@ CNode::CNode(NodeId idIn, ServiceFlags nLocalServicesIn, int nMyStartingHeightIn
     // Don't relay addr messages to peers that we connect to as block-relay-only
     // peers (to prevent adversaries from inferring these links from addr
     // traffic).
-    m_addr_relay_peer(!block_relay_only),
+    m_addr_known{block_relay_only ? nullptr : MakeUnique<CRollingBloomFilter>(5000, 0.001)},
     id(idIn),
     nLocalHostNonce(nLocalHostNonceIn),
     nLocalServices(nLocalServicesIn),
@@ -2682,10 +2682,6 @@ CNode::CNode(NodeId idIn, ServiceFlags nLocalServicesIn, int nMyStartingHeightIn
         m_tx_relay = MakeUnique<TxRelay>();
     }
 
-    if (m_addr_relay_peer) {
-        m_addr_known = MakeUnique<CRollingBloomFilter>(5000, 0.001);
-    }
-
     for (const std::string &msg : getAllNetMessageTypes())
         mapRecvBytesPerMsgCmd[msg] = 0;
     mapRecvBytesPerMsgCmd[NET_MESSAGE_COMMAND_OTHER] = 0;
diff --git a/src/net.h b/src/net.h
index 327c06a306..d71e30377a 100644
--- a/src/net.h
+++ b/src/net.h
@@ -776,13 +776,12 @@ public:
 
     // flood relay
     std::vector<CAddress> vAddrToSend;
-    std::unique_ptr<CRollingBloomFilter> m_addr_known;
+    const std::unique_ptr<CRollingBloomFilter> m_addr_known;
     bool fGetAddr{false};
     int64_t nNextAddrSend GUARDED_BY(cs_sendProcessing){0};
     int64_t nNextLocalAddrSend GUARDED_BY(cs_sendProcessing){0};
 
-    const bool m_addr_relay_peer;
-    bool IsAddrRelayPeer() const { return m_addr_relay_peer; }
+    bool IsAddrRelayPeer() const { return bool{m_addr_known}; }
 
     // List of block ids we still have announce.
     // There is no final sorting before sending, as they are always sent immediately